docs/en/stack/ml/anomaly-detection/ml-troubleshooting.asciidoc

```diff
@@ -1,118 +1,154 @@
+*Anomaly Explorer* fails to load.
+* The following error occurs in {kib} when you try to view annotations for
+{anomaly-jobs}: `Error loading the list of annotations for this job`
+* Cannot create or update any {ml} jobs. The error messages in this case are
+illegal argument exceptions like `mapper [model_plot_config.annotations_enabled]
+cannot be changed from type [keyword] to [boolean]`. This problem is most likely
+to occur if after upgrading you open an existing {anomaly-job} in 7.9.0 before
+you create or update a job. 
 
 *Resolution:*
 
-Create {anomaly-jobs} in {kib} or ensure that you create {anomaly-jobs} with
-valid identifiers when you use the APIs. For more information about valid
-identifiers, see
-{ref}/ml-put-job.html[Create {anomaly-jobs} API].
-
-[[ml-upgradedf]]
-== Upgraded nodes fail to start due to {dfeed} issues
-
-This problem occurs when you have a {dfeed} that contains search or query
-domain specific language (DSL) that was discontinued. For example, if you
-created a {dfeed} query in 5.x using search syntax that was deprecated in 5.x
-and removed in 6.0, you must fix the {dfeed} before you upgrade to 6.0.
-
-*Symptoms:*
-
-* If {ref}/logging.html#deprecation-logging[deprecation logging] is enabled
-before the upgrade, deprecation messages are generated when the {dfeeds} attempt
-to retrieve data.
-* After the upgrade, nodes fail to start and the error indicates that they
-failed to read the local state.
-
-*Resolution:*
-
-Before you upgrade, identify the problematic search or query DSL. In 5.6.5 and
-later, the Upgrade Assistant detects these scenarios. If you cannot fix the DSL
-before the upgrade, you must delete the {dfeed} then re-create it with valid DSL
-after the upgrade.
-
-If you do not fix or delete the {dfeed} before the upgrade, in order to successfully
-start the failing nodes you must downgrade the nodes then fix the problem per
-above.
-
-See also {stack-ref}/upgrading-elastic-stack.html[Upgrading the Elastic Stack].
\ No newline at end of file
+To avoid this problem, manually update the mappings on the {ml} annotations and
+config indices in your old {es} version before you upgrade to 7.9.0. For example:
+
+[source,console]
+--------------------------------------------------
+PUT .ml-annotations-6/_mapping
+{
+  "properties": {
+    "event" : {
+      "type" : "keyword"
+    },
+    "detector_index" : {
+      "type" : "integer"
+    },
+    "partition_field_name" : {
+      "type" : "keyword"
+    },
+    "partition_field_value" : {
+      "type" : "keyword"
+    },
+    "over_field_name" : {
+      "type" : "keyword"
+    },
+    "over_field_value" : {
+      "type" : "keyword"
+    },
+    "by_field_name" : {
+      "type" : "keyword"
+    },
+    "by_field_value" : {
+      "type" : "keyword"
+    }
+  }
+}
+--------------------------------------------------
+// TEST[skip:TBD]
+
+//TBD: Are the same mappings required for the .ml-config index?
```

Comment:
The mapping additions for the config index are in https://github.com/elastic/elasticsearch/issues/61157#issuecomment-674143895

```
PUT .ml-config/_mapping
{
  "properties": {
    "analysis_config": {
      "properties": {
        "per_partition_categorization" : {
          "properties" : {
            "enabled" : {
              "type" : "boolean"
            },
            "stop_on_warn" : {
              "type" : "boolean"
            }
          }
        }
      }
    },
    "max_num_threads" : {
      "type" : "integer"
    },
    "model_plot_config" : {
      "properties" : {
        "annotations_enabled" : {
          "type" : "boolean"
        }
      }
    }
  }
}
```